app/services/foreman_ansible/roles_importer.rb

```diff
@@ -1,24 +1,54 @@
 module ForemanAnsible
-  # Service to import roles from the filesystem.
-  # Should be extracted to a gem that both the proxy and Foreman can use.
   class RolesImporter
-    class << self
-      def import!
-        list_roles_in_fs.each do |role_name|
-          role = AnsibleRole.new(:name => role_name)
-          next if role.save
-          Rails.logger.debug("Failed to save role #{role_name}: "\
-                            "#{role.errors.full_messages.join(', ')}")
-        end
+    attr_reader :ansible_proxy
+
+    def initialize(proxy = nil)
+      @ansible_proxy = proxy
+    end
+
+    def proxy_api
```

Comment:
I could, but import is two step process. In the first step (`import` action), we get the roles from proxy, display the changes to user and let him decide what to import. In the second step (`confirm_import` action), we simply persist the changes user selected. We do not actually need proxy for the second step. We could also move `@proxy_api = find_proxy_api` into initializer and extract the `finish_import` method into a separate class.